pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -44,19 +47,137 @@ function subvolume_mount(volume, subvol, forced_options) {
     mounting_dialog(client, block, "mount", forced_options, subvol);
 }
 
+function get_mount_point_in_parent(volume, subvol) {
+    const block = client.blocks[volume.path];
+    const subvols = client.uuids_btrfs_subvols[volume.data.uuid];
+    if (!subvols)
+        return null;
+
+    for (const p of subvols) {
+        const has_parent_subvol = subvol.pathname.substring(0, p.pathname.length) == p.pathname &&
+                                 subvol.pathname[p.pathname.length] == "/";
+        if ((subvol.parent == p.id || has_parent_subvol) && is_mounted(client, block, p)) {
+            const [, pmp, opts] = get_fstab_config_with_client(client, block, false, p);
+            const opt_ro = extract_option(parse_options(opts), "ro");
+            if (!opt_ro) {
+                if (p.pathname == "/")
+                    return pmp + "/" + subvol.pathname;
+                else
+                    return pmp + subvol.pathname.substring(p.pathname.length);
+            }
+        }
+    }
+    return null;
+}
+
+function set_mount_options(subvol, block, vals) {
+    const mount_options = [];
+    const mount_now = vals.variant != "nomount";
+
+    if (!mount_now || vals.at_boot == "never") {
+        mount_options.push("noauto");
+    }
+    if (vals.mount_options.ro)
+        mount_options.push("ro");
+    if (vals.at_boot == "never")
+        mount_options.push("x-cockpit-never-auto");
+    if (vals.at_boot == "nofail")
+        mount_options.push("nofail");
+    if (vals.at_boot == "netdev")
+        mount_options.push("_netdev");
+
+    const name = (subvol.pathname == "/" ? vals.name : subvol.pathname + "/" + vals.name);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19849-20240118-154511-b2bc543c-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)